client/spec/components/Avatar.spec.jsx

```diff
@@ -1,46 +1,38 @@
-import avatarFactory from "inject!Avatar";
-import ReactDOM from "react-dom";
+import Avatar from "Avatar";
 
 describe( "Avatar", () => {
-	let component, dependencies;
+	let component, container;
 
 	beforeEach( () => {
-		dependencies = {};
-
-		const Avatar = avatarFactory( dependencies );
-
-		component = ReactUtils.renderIntoDocument( <Avatar /> );
+		container = document.createElement( "div" );
 	} );
 
-	afterEach( () => {
-		if ( component ) {
-			ReactDOM.unmountComponentAtNode( ReactDOM.findDOMNode( component ).parentNode );
-		}
-	} );
+	function render( props ) {
+		ReactDOM.render( <Avatar {...props} />, container );
+		component = container.children[ 0 ];
```

Comment:
`node = ReactDOM.findDOMNode( null );` so that won't work here. Feel free to pull the code down and try out some ideas. I don't think I was missing anything, but it behaved quite differently during tests, so I ended up with this setup to work around the restrictions.